### PR TITLE
Remove sscanf and sprintf usage

### PR DESCRIPTION
**Pull request by janvorli**
This change removes sscanf and sprintf implementation and usage from coreclr and PAL. The usage is replaced by the safe counterparts sscanf_s, sprintf_s and _snprintf_s. The sprintf_s and _snprintf_s is chosen based on whether the resulting string is bounded in size or not.

### CODE REVIEW

**Comment by janvorli** [comment]
@adityamandaleeka or @jkotas can you take a look please?
@BruceForstall can you please take a look at the changes in JIT and superpmi?

File: src/pal/tests/palsuite/c_runtime/sscanf_s/test10/test10.cpp [hunk 1]
```diff
@@ -6,15 +6,15 @@
 **
 ** Source:  test10.c
 **
-** Purpose:  Tests sscanf with wide charactersn
+** Purpose:  Tests sscanf_s with wide charactersn
```
Comment (adityamandaleeka) [inline src/pal/tests/palsuite/c_runtime/sscanf_s/test10/test10.cpp]: charactersn -> characters ?

**Comment by adityamandaleeka** [comment]
LGTM

**Comment by janvorli** [comment]
Two of the PAL tests fail on FreeBSD. I need to investigate what's wrong.

**Comment by janvorli** [comment]
@mmitche what is the version of FreeBSD and installed packages we are using in the CI? I am unable to repro the PAL test failure I am hitting in the CI on my local FreeBSD 10.1 VM.

**Comment by mmitche** [comment]
Hmm..I think it's 10.1.  I'll rerun and pull the machine offline for investigation.

**Comment by mmitche** [comment]
@dotnet-bot test FreeBSD x64 Checked Build

**Comment by janvorli** [comment]
@adityamandaleeka I have found the issue that has caused the FreeBSD failure. It was actually a bug in the two tests - I have not realized that sscanf_s requires passing additional buffer size parameter for %C and %c formats. It was passing on Linux by a mere chance.
I have also fixed the comment and one benign usage of L'?' instead of W('?').
Can you take a look again please?

**Comment by adityamandaleeka** [comment]
LGTM

**Comment by BruceForstall** [comment]
@janvorli sorry I didn't have a chance to look earlier; the changes look fine to me.

**Pull request merged**